### DISCUSSION TITLE
How to access to an enviroment variable in vercel.json

### DISCUSSION THREAD

**Discussion by elVengador** [vercel/community#5074] [Help]
### Summary

I want to protect a value on my vercel.json set up
But it is not working, any suggestion?

### Steps to Reproduce

I set an environment variable ALLOW_ORIGIN in Vercel page > settings > environment variables, and I've tried to access in this way in my vercel.json:
```
{
    "version": 2,
    "builds": [
      {
        "src": "index.js",
        "use": "@now/node"
      }
    ],
    "headers": [
      {
        "source": "/",
        "headers": [
          { "key": "Access-Control-Allow-Origin", "value": "${ALLOW_ORIGIN}" },
      }
    ]
  }
```

**Comment by amyegan** [comment]
Hey @elVengador. Environment variables cannot be used in `vercel.json`, but there are a couple of other options.

You can [modify request headers](https://vercel.com/templates/next.js/edge-functions-modify-request-header) with [Edge Middleware](https://vercel.com/docs/functions/edge-middleware). Or, if your project uses Next.js, you can [use next.config.js to set headers](https://nextjs.org/docs/pages/api-reference/next-config-js/headers).

I also noticed that you're using the [`builds`](https://vercel.com/docs/projects/project-configuration#builds) option and `@now/node`. We recommend using the [`functions`](https://vercel.com/docs/projects/project-configuration#functions) property to customize Serverless Functions instead. Short of that, I recommend at least updating from `@now/node` to `@vercel/node`.
